curvefs/src/mds/topology/topology_metric.cpp

```diff
@@ -83,6 +89,15 @@ void TopologyMetricService::UpdateTopologyMetrics() {
         for (auto pit = partitions.begin(); pit != partitions.end(); ++pit) {
             totalInodeNum += pit->GetInodeNum();
             totalDentryNum += pit->GetDentryNum();
+            // update fs inodeNum metrics
+            auto fsId = pit->GetFsId();
+            auto it = gFsMetrics.find(fsId);
+            if (it == gFsMetrics.end()) {
+                FsMetricPtr fptr(new FsMetric(fsId));
+                it = gFsMetrics.emplace(fsId, std::move(fptr)).first;
+            }
+            it->second->inodeNum_.set_value(pit->GetInodeNum() +
+                                            it->second->inodeNum_.get_value());
```

Comment:
I think you can use a map<FsId *fsid*, uint64_t *total inode number*> or unordered_map to record the intermedia value when looping through all the partitions, and after that, call set_value for each fs once.

Reply:
> I think you can use a map<FsId _fsid_, uint64_t _total inode number_> or unordered_map to record the intermedia value when looping through all the partitions, and after that, call set_value for each fs once.

fix